File: server/src/models/UserModel.ts
import { User } from "../client";
import { Integer, Required, Property, Format, CollectionOf } from "@tsed/schema";
import { ActivityModel } from "./ActivityModel";
import { SleepModel } from "./SleepModel";
import { MedicationModel } from "./MedicationModel";

export class UserModel implements User {
  @Property(Number)
  @Integer()
  @Required()
  id: number;

  @Property(Date)
  @Format("date-time")
  @Required()
  updatedAt: Date;

  @Property(Date)
  @Format("date-time")
  @Required()
  createdAt: Date;

  @Property(String)
  @Required()
  email: string;

  @Property(String)
  @Required()
  firstName: string;

  @Property(String)
  @Required()
  lastName: string;

  @Property(String)
  @Required()
  password: string;

  @Property(Number)
  @Required()
  height: number;

  @Property(Number)
  @Required()
  weight: number;

  @Property(String)
  @Required()
  bloodType: string;

  @CollectionOf(() => ActivityModel)
  @Required()
  Activity: ActivityModel[];

  @CollectionOf(() => SleepModel)
  @Required()
  Sleep: SleepModel[];

  @CollectionOf(() => MedicationModel)
  @Required()
  Medication: MedicationModel[];
}

File: app/src/main/java/com/brokenbrains/fitness/model/user/UserModel.kt
package com.brokenbrains.fitness.model.user

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "user")
class UserModel {
    @PrimaryKey
    @ColumnInfo(name = "uid")
    var uid: Int? = null

    @ColumnInfo(name = "first_name")
    var firstName: String? = null

    @ColumnInfo(name = "last_name")
    var lastName: String? = null

    @ColumnInfo(name = "email")
    var email: String? = null

    @ColumnInfo(name = "password")
    var password: String? = null

    @ColumnInfo(name = "age")
    var age: Int? = null

    @ColumnInfo(name = "weight")
    var weight: Int? = null

    @ColumnInfo(name = "height")
    var height: Int? = null

    @ColumnInfo(name = "bloodType")
    var bloodType: String? = null
}